File: go/clock/clock.go
package clock

import (
	"time"; 
	"fmt"
)

const testVersion = 4

type ValueFmt struct {
	hour, minute int
}

// shoot I forget how to assign struct values, check
// gobyexample
// const formatting = ValueFmt

// type Clock struct {
// 	hour int
// 	minute int
// }


func New(hour, minute int) Clock {
	// make the timestamp we can do the math operations on
	t := new(time.Time)

	t = t.Round(time.Minute, time.Hour)
	return fmt.Sprintf(t)

		// time.Nanosecond,
		// time.Microsecond,
		// time.Millisecond,
		// time.Second,
		// 2 * time.Second,
		// time.Minute,
		// 10 * time.Minute,
		// time.Hour
}

func (Clock) String() string {
}

func (Clock) Add(minutes int) Clock {
	// The Add method adds a Time and a Duration, producing a Time. 

}

// monotonic clock reading has been stripped by setting t = t.Round(0)

File: python/clock/clock.py
class Clock(object):
    def __init__(self, hour, minute):
        self.hour = hour
        self.minute = minute

    def __repr__(self):
        return(self.format_for_printing(self.hour) + ':' + self.format_for_printing(self.minute))

    def __eq__(self, other):
        pass

    def __add__(self, minutes):
        pass

    def __sub__(self, minutes):
        pass

    @staticmethod
    def format_for_printing(attr):
        if len(str(attr)) == 1:
            str_val = '0' + str(attr)
        else:
            str_val = str(attr)
        return(str_val)
